openfl-tutorials/interactive_api_tutorials_(experimental)/Pytorch_TinyImageNet_workspace/new_python_api_tinyimagenet.ipynb

```diff
@@ -0,0 +1,670 @@
+    "                for cnt in range(NUM_IMAGES_PER_CLASS):\n",
+    "                    self.labels[f'{label_text}_{cnt}.JPEG'] = i\n",
+    "        elif split == 'val':\n",
+    "            with open(os.path.join(self.split_dir, 'val_annotations.txt'), 'r') as fp:\n",
+    "                for line in fp.readlines():\n",
+    "                    terms = line.split('\\t')\n",
+    "                    file_name, label_text = terms[0], terms[1]\n",
+    "                    self.labels[file_name] = self.label_text_to_number[label_text]\n",
+    "                    \n",
+    "    \n",
+    "    def __len__(self):\n",
+    "        return len(self.image_paths)\n",
+    "\n",
+    "    def __getitem__(self, index):\n",
+    "        file_path = self.image_paths[index]\n",
+    "        label = self.labels[os.path.basename(file_path)]\n",
+    "        label = self.target_transform(label) if self.target_transform else label\n",
+    "        return self.read_image(file_path), label\n",
+    "\n",
+    "    def read_image(self, path):\n",
+    "        img = Image.open(path)\n",
+    "        return self.transform(img) if self.transform else img\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4ec1f1ca",
+   "metadata": {},
+   "source": [
+    "### Define Federated Learning tasks"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "79d9ec24",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def train(net_model, train_loader, optimizer, device, loss_fn=cross_entropy):\n",
+    "    \n",
+    "    function_defined_in_notebook()\n",
+    "    \n",
+    "    net_model.train()\n",
+    "    net_model.to(device)\n",
+    "\n",
+    "    losses = []\n",
+    "\n",
+    "    for data, target in train_loader:\n",
+    "        data, target = torch.tensor(data).to(device), torch.tensor(\n",
+    "            target).to(device, dtype=torch.float32)\n",
+    "        optimizer.zero_grad()\n",
+    "        output = net_model(data)\n",
+    "        loss = loss_fn(output=output, target=target)\n",
+    "        loss.backward()\n",
+    "        optimizer.step()\n",
+    "        losses.append(loss.detach().cpu().numpy())\n",
+    "        \n",
+    "    return {'train_loss': np.mean(losses),}\n",
+    "\n",
+    "\n",
+    "def validate(net_model, val_loader, device):\n",
+    "    net_model.eval()\n",
+    "    net_model.to(device)\n",
+    "\n",
+    "    val_score = 0\n",
+    "    total_samples = 0\n",
+    "\n",
+    "    with torch.no_grad():\n",
+    "        for data, target in val_loader:\n",
+    "            samples = target.shape[0]\n",
+    "            total_samples += samples\n",
+    "            data, target = torch.tensor(data).to(device), \\\n",
+    "                torch.tensor(target).to(device, dtype=torch.int64)\n",
+    "            output = net_model(data)\n",
+    "            val = cross_entropy(output, target)\n",
```

Comment:
We don't usually measure the Cross-Entropy value during the validation task of the classification problem. Accuracy is more relevant here. See how we get the 'acc' metric in the `validate` function in [openfl/federated/task/runner_pt.py](https://github.com/intel/openfl/blob/tinyimagenet/openfl/federated/task/runner_pt.py)

Reply:
Thanks Ilya! I followed the metric mentioned in the TinyImagenet tutorial from the old API to be consistent. Should we change the metric in both cases then?

Reply:
Do you mean [openfl-tutorials/Federated_PyTorch_TinyImageNet.ipynb](https://github.com/intel/openfl/blob/tinyimagenet/openfl-tutorials/Federated_PyTorch_TinyImageNet.ipynb)? Network class there doesn't contain definition of `train_batches` or `validate` functions, so default `PyTorchTaskRunner` implementations are used.